### ISSUE TITLE
Telegraf won't start on empty (zero-length) statefile

### ISSUE THREAD

**Issue by burkart** [influxdata/telegraf#14935]
### Relevant telegraf.conf

```toml
[agent]
  statefile = "/var/lib/telegraf/statefile"

# Dedup uses the statefile
[[processors.dedup]]
  dedup_interval = "60s"
```

### Logs from Telegraf

```text
E! [telegraf] Error running agent: unmarshalling states failed: unexpected end of JSON input
```

### System info

Telegraf 1.29.2, Debian 12.4, Docker 24.0.7

### Steps to reproduce

1. Enable statefile in telegraf.conf
2. Make sure statefile is empty (0 bytes)
3. Start Telegraf

### Expected behavior

Telegraf starts and fills statefile on exit.

### Actual behavior

Telegraf fails to start.

### Additional info

Telegraf comes with a zero-length statefile /var/lib/telegraf/statefile that's generated by scripts/{deb,rpm}/post-install.sh.

Telegraf starts successfully if the file contents are "{}" (without the quotes).

**Comment by shengbinxu** [comment]
Yes, I also encountered this issue. The version I am using is 1.29.1, and the issue can be resolved by writing the following content in the file.
```
{}
```

**Comment by srebhan** [comment]
@shengbinxu this should be fixed in v1.30.0+...
